utils/collection-concepts.hh

```diff
@@ -41,13 +42,18 @@ SEASTAR_CONCEPT(
     concept LessNothrowComparable = LessComparable<Key1, Key2, Less> && std::is_nothrow_invocable_v<Less, Key1, Key2>;
 )
 
+// Temporary scaffolding for supporting trichotomic compares
+// that return either int or std::strong_ordering
+template <typename T>
+concept int_or_strong_ordering = std::same_as<T, int> || std::same_as<T, std::strong_ordering>;
```

Comment:
I'm a little unsure on what our coding style is - what's the point of using the "concept" keyword here without any old-compiler protection, and still use the SEASTAR_CONCEPT macro two lines below?
If the decision is that we don't need SEASTAR_CONCEPT any more, maybe you should remove it below? I know it's preexisting, but since you're adding a new concept anyway, it's a very good time.

Reply:
Scylla requires C++20, which comes with concepts.

I'm happy to follow up with a patch that removes SEASTAR_CONCEPT globally, but would rather not touch unrelated code in this patch.

Reply:
Well, I obviously don't expect you to remove SEASTAR_CONCEPT globally, but here you are specifically creating a new concept to be used one line down, and yet the one line down is "protected" by SEASTAR_CONCEPT, but the new concept isn't. I think it's a good opportunity to at least clean this line. The mixture in the current patch is just strange.

Reply:
I don't see how the patch is mixing anything. It's certainly not our style to have drive-by fixes.

Reply:
Well, unless I'm misreading it, your patch takes code which supports old compilers (because it's "protected" by SEASTAR_CONCEPT) and then actively breaks this support by adding a new concept outside the SEASTAR_CONCEPT. I think you need to decide whether you want to keep this support (and hide the new concept as well) or if you don't want to keep it, but the combination doesn't make sense.

Reply:
https://github.com/scylladb/scylla/pull/8236